src/liboslexec/CMakeLists.txt

```diff
@@ -111,7 +111,7 @@ macro ( LLVM_COMPILE llvm_src srclist )
           "-I${ILMBASE_INCLUDES}"
           "-isystem ${Boost_INCLUDE_DIRS}"
           -DOSL_COMPILING_TO_BITCODE=1
-          -Wno-deprecated-register
+          -Wno-deprecated-register -Wno-ignored-attributes -Wno-unknown-attributes
```

Comment:
Rather than suppressing all warnings about unknown or ignored attributes, I wonder if the better fix is to modify export.h so that in the Cuda case, they are not added. That is, change this line (45)

    #    if defined(OSL_STATIC_BUILD) || defined(OSL_STATIC_DEFINE)

by adding `|| defined(__CUDA_ARCH__)`

???

Reply:
HI, yes you're absolutely right, and that's how I started fixing this warning. But this needs to be done in OIIO and OpenEXR at least (I don't remember exactly how many 3rd parties I had to patch like this before reverting and going the `-Wno-xx` way)
But yep, I should probably revert this, fix OSL's export macro, and submit PRs to 3rd parties.

Reply:
I see, of course. I wasn't thinking about the dependency headers.

Yes, I think that the best thing is:
* make the best changes to OSL's export macros (at the very least, so we can show by example how it should be done)
* submit PRs to the 3rd parties when you have a chance
* keep the warning suppression, but maybe put a comment somewhere nearby in that cmake file saying that the particular reason -Wno-x and -Wno-y are needed is because of those dllimport decorators and that the warnings can be restored once we're sure that any upstream projects have fixed their headers and we are guaranteed to be using at something newer than those fixed versions.